posthog/tasks/session_recordings.py

```diff
@@ -0,0 +1,129 @@
+from celery import shared_task
+from celery.app.task import Task
+from posthog.models import Team, User
+from posthog.session_recordings.models.session_recording import SessionRecording
+from posthog.session_recordings.models.session_recording_playlist_item import SessionRecordingPlaylistItem
+from posthog.schema import RecordingsQuery
+from posthog.event_usage import report_user_action
+import structlog
+from typing import Any
+
+logger = structlog.get_logger(__name__)
+
+
+@shared_task(bind=True)
+def bulk_delete_recordings_task(
+    self: Task, team_id: int, user_id: int, filters: dict[str, Any], user_distinct_id: str
+) -> dict[str, Any]:
+    """
+    Bulk delete recordings matching the provided filters.
+    Also mark associated playlist items as deleted.
+    Processed in batches to avoid memory issues.
+    """
+    from posthog.session_recordings.session_recording_api import list_recordings_from_query
+
+    try:
+        team = Team.objects.get(id=team_id)
+        user = User.objects.get(id=user_id)
+        query = RecordingsQuery.model_validate(filters)
+
+        CHUNK_SIZE = 100
+        deleted_count = 0
+        playlist_items_deleted_count = 0
+        offset = 0
+
+        # Process recordings in chunks using database pagination
+        while True:
+            # Set pagination for this chunk
+            query.limit = CHUNK_SIZE
+            query.offset = offset
+
+            # Get chunk of recordings
+            chunk_recordings, _, _ = list_recordings_from_query(query, user, team)
+
+            if not chunk_recordings:
+                break  # No more recordings to process
+
+            session_ids_chunk = [str(r.session_id) for r in chunk_recordings]
+
+            # 1. Update existing Postgres SessionRecording records
+            SessionRecording.objects.filter(team=team, session_id__in=session_ids_chunk).update(deleted=True)
+
+            # 2. Create recording for ClickHouse-only recordings
+            existing_session_ids = set(
```

Comment:
you can upsert with django orm

https://docs.djangoproject.com/en/5.2/ref/models/querysets/#update-or-create

so can do this in one step (I think)